private/buf/cmd/buf/buf.go

```diff
@@ -317,14 +317,29 @@ func wrapError(err error) error {
 		connectCode := connectErr.Code()
 		switch {
 		case connectCode == connect.CodeUnauthenticated, isEmptyUnknownError(err):
-			if authErr, ok := bufconnect.AsAuthError(err); ok && authErr.TokenEnvKey() != "" {
-				return fmt.Errorf("Failure: the %[1]s environment variable is set, but is not valid. "+
-					"Set %[1]s to a valid Buf API key, or unset it. For details, "+
-					"visit https://docs.buf.build/bsr/authentication", authErr.TokenEnvKey())
+			if authErr, ok := bufconnect.AsAuthError(err); ok {
+				if authErr.HasToken() && authErr.TokenEnvKey() != "" {
+					return fmt.Errorf("Failure: the %[1]s environment variable is set, but is not valid. "+
+						"Set %[1]s to a valid Buf API key, or unset it. For details, "+
+						"visit https://buf.build/docs/bsr/authentication",
+						authErr.TokenEnvKey(),
+					)
+				}
+				if authErr.HasToken() {
+					return fmt.Errorf("Failure: invalid token for %s found in netrc. "+
+						"Set it to a valid Buf API key. For details, visit https://buf.build/docs/bsr/authentication",
+						authErr.Remote(),
+					)
+				}
+				return fmt.Errorf("Failure: you are not authenticated for %s. Create a new entry in your netrc, "+
```

Comment:
This doesn't fix #2500 - #2500 states that you should also talk about the environment variable, not just the netrc file, but this does nothing to address that

Reply:
I think [2893b46](https://github.com/bufbuild/buf/pull/3093/commits/2893b46feeffedab38a253d82154c1c41a5a3911) resolves it

Reply:
yep!